compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -203,7 +213,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                                 // We expected a lifetime argument, but got a type or const
                                 // argument. That means we're inferring the lifetimes.
                                 substs.push(ctx.inferred_kind(None, param, infer_args));
-                                force_infer_lt = Some(arg);
+                                force_infer_lt = Some((arg, param));
```

Comment:
is it possible that this loop is a lot hotter than I expected so storing arg with param here is quite costly?

Other than that i really don't know waht could be causing the perf regression.

Reply:
I agree that this would be the most likely cause, it could be that the type for param for deeply-nested-async stuff is huge and thus takes up a lot of space on the stack, so even if it's not hot it might just be causing a lot more cache misses

Reply:
It seems possible that this increases the size of force_infer_lt enough that it becomes stored on the stack instead of a register, I guess, but it's pretty odd that it would cause any problems (still just two pointers vs one). An index would still be a usize so should be equally costly I'd expect.

Reply:
I added a print to the inner body, and during compilation it appears that the body is hit at least 1.5M times, so I guess that's fairly hot?

Reply:
you mean during the compilation of the compiler itself? so during `./x.py build --stage 2` or just while building std?

Reply:
I was running `./x.py test src/test/ui`, but it got to compiling stage 1 artifacts and I quit out